### ISSUE TITLE
Low accuracy for pose estimation?

### ISSUE THREAD

**Issue by PatrickBue** [opencv/open_model_zoo#27]
Hi,
In the description of human-pose-estimation-0001.md it states that the mean Average Precision on Coco is 42.8%. In the OpenPose paper, the authors report an mAP of around 60%. Does this mean that the current implementation is not as accurate?
Thanks,

**Comment by snosov1** [comment]
`human-pose-estimation-0001` is a much smaller model (compared to original OpenPose) that can run in real-time on fairly modest HW. The accuracy numbers are, indeed, smaller than the original - however, it might be enough for your practical applications, so we urge you to give it a try and see if it works for you.

That is the case for most of our models. We apply something like an 80-20 rule. Typically, you can get 80% of the possible accuracy with just 20% of the compute. To get the "last" 20% you need to make your model 5 times slower. If you really want the highest accuracy possible (no matter the cost), then you can use [other implementations](https://github.com/CMU-Perceptual-Computing-Lab/openpose)

cc @daniil-osokin - model author

**Comment by PatrickBue** [comment]
That makes sense. Thanks a lot for the super quick reply, and might be a good idea to add this explanation to the human-pose-estimation-0001.md file.

**Comment by lunalulu** [comment]
can you provide me with a well-trained model(the lightweight openpose, .caffemodel)??
thank you very much!

**Comment by Daniil-Osokin** [comment]
Pre-trained [model](https://software.intel.com/en-us/openvino-toolkit/documentation/pretrained-models) is available in OpenVINO format. It can be found inside of [OpenVINO Toolkit](https://software.intel.com/en-us/openvino-toolkit/choose-download), as well as corresponding demo app.

**Comment by lunalulu** [comment]
thanks for your reply!
Now I have some questions:
1、on OpenVINO™ 2018 R5 Release version，Compile samples and report an error as follows:
`
i.Navigate to a directory that you have write access to and create a samples build directory. This example uses a directory named build: mkdir build;
ii.cd <path_to_build_directory>
iii.cmake -DCMAKE_BUILD_TYPE=Release <INSTALL_DIR>/deployment_tools/inference_engine/samples/

error: could not find a package configuration file provided by "inferenceEngine"
`
2、i Using OpenCV DNN to achieve human pose detection(human-pose-estimation-0001), but also can not run, FP32 on cpu report illegal  instruction(core dumped); FP16 on NCS2 report can not init USB device:NC_DEVICE_NOT_FOUND; I configured the environment variables;

3、when i run my code on  OpenVINO™ 2018 R4 Release version with its human-pose-estimation-0001 model, it alse report error:
segmention fault

**Comment by lunalulu** [comment]
i found the problom,but i dont know how to do, my opencv code as follows:

this->blob = blobFromImage(img, 1.0/255.0, Size(456,256), Scalar(0,0,0), false, false);
this->net.setInput(this->blob);
Mat netOutputBlob = this->net.forward();

i found the size of the forward output  netOutputBlob is 19, not 57~

please tell me how to get the right output?? thank you very much~

**Comment by lunalulu** [comment]
now i can get human-pose-estimation-0001 two outputblobs, as follows:
    Mat netOutputBlobHeatMap = this->net.forward("Mconv7_stage2_L2");
    Mat netOutputBlobPaf = this->net.forward("Mconv7_stage2_L1");

but still report segmentation fault

It looks like multi outputs will give you an error of segmentation fault

**Comment by dkurt** [comment]
@lunalulu, try to use OpenCV's samples:

* [Python](https://github.com/opencv/opencv/blob/master/samples/dnn/openpose.py) 
* [C++](https://github.com/opencv/opencv/blob/master/samples/dnn/openpose.cpp)

example:

```
python ../samples/dnn/openpose.py \
  --proto /opt/intel/computer_vision_sdk_2018.5.445/deployment_tools/intel_models/human-pose-estimation-0001/FP32/human-pose-estimation-0001.xml \
  --model /opt/intel/computer_vision_sdk_2018.5.445/deployment_tools/intel_models/human-pose-estimation-0001/FP32/human-pose-estimation-0001.bin \
  --width 456 --height 256 \
  --dataset COCO \
  --scale 1.0
```

**Comment by Daniil-Osokin** [comment]
@lunalulu Since samples need some environment variables set, I suggest to run `./build_samples.sh`, which is inside samples folder to build them.

**Comment by lunalulu** [comment]
@Daniil-Osokin do you run the human-pose-estimation-0001 using opencv API ? can it return right results?

**Comment by lunalulu** [comment]
@dkurt i will try your method, thanks
i want to know do you run the human-pose-estimation-0001 using opencv API ? can it return right results?

**Comment by Daniil-Osokin** [comment]
Yes, I've run OpenCV C++ sample with our model. The only lines I've changed is set scale to 1.0 and changed `readNetFromCaffe` to `readNetFromModelOptimizer` (it's for OpenCV 4.0.0/4.0.1, in master this fixed). Please note, that this sample works good for images with 1 person (as noted in comments), for multi-person case use OpenVINO demo.

**Comment by snosov1** [comment]
@lunalulu is this still relevant or we can close?

**Comment by lunalulu** [comment]
@snosov1 sorry，I don't have time to verify it, but I think you can close it first! thanks very much!

**Comment by snosov1** [comment]
Sure, feel free to reopen if this becomes relevant again! Thx!
